app/services/delete_account_service.rb

```diff
@@ -178,6 +178,23 @@ def purge_generated_notifications!
     Notification.where(from_account: @account).in_batches.delete_all
   end
 
+  def purge_favourites!
+    @account.favourites.in_batches do |favourites|
+      ids = favourites.pluck(:status_id)
```

Comment:
Here’s the thing, internally in_batches plucks the favourite IDs to then construct the relation from which then you pluck these. Feels like there should be a way to do it in one query instead of two...

Reply:
`status_id` ≠ `id`

Anyway, yeah, the underlying query could have included `status_id` and then we wouldn't have to do two queries, but I can't seem to find how to do that with Rails and it's not a big issue anyway.